Zend/zend_alloc.c

```diff
@@ -1432,8 +1519,8 @@ static zend_always_inline void zend_mm_free_small(zend_mm_heap *heap, void *ptr,
 #endif
 
 	p = (zend_mm_free_slot*)ptr;
-	zend_mm_set_next_free_slot(heap, bin_num, p, heap->free_slot[bin_num]);
-	heap->free_slot[bin_num] = p;
+	zend_mm_set_next_free_slot(heap, bin_num, p, ZEND_MM_FREE_SLOT_EX(heap, chunk, bin_num));
+	ZEND_MM_FREE_SLOT_EX(heap, chunk, bin_num) = p;
```

Comment:
This make regression for each deallocation.

Reply:
This adds an additional fetch of `chunk->zone_free_slot`, but this field is just next to `chunk->heap`, and is likely in cache as we fetch it here:

https://github.com/php/php-src/blob/53df3ae1e531f6f9ee9801a8340c7c0033bc57e7/Zend/zend_alloc.c#L2701-L2702